src/test/java/api/support/APITests.java

```diff
@@ -386,4 +387,11 @@ protected void mockClockManagerToReturnFixedDateTime(DateTime dateTime) {
   protected void mockClockManagerToReturnDefaultDateTime() {
     ClockManager.getClockManager().setDefaultClock();
   }
+
+  protected void assertThatSentNoticesCountIsEqualToLogRecordEventsCount() {
+    int logRecordEventsCount = FakePubSub.getPublishedEvents().stream()
+      .filter(json -> "LOG_RECORD_EVENT".equals(json.getString("eventType")))
+      .collect(Collectors.reducing(0, e -> 1, Integer::sum));
+    assertThat(patronNoticesClient.getAll().size(), equalTo(logRecordEventsCount));
+  }
```

Comment:
The APITests class already has a lot of different concerns, could this be moved into a separate class?